core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -224,6 +229,12 @@ public void initialize(String name, Map<String, String> unresolved) {
               client, tokenRefreshExecutor(), token, expiresAtMillis(mergedProps), catalogAuth);
     }
 
+    String restPaginationEnabledValue = mergedProps.get(REST_PAGINATION_ENABLED);
```

Comment:
Do we need the feature flag? We designed these changes in a way that makes the pagination backwards compatible. Therefore, if the service response includes more results than the page size it should still work. Then we can simplify the logic in the listing methods to handle pagination but still accommodate for services sending  all results in single response.

What do you think?

Reply:
+1, I think we don't need the feature flag. 

I see the pagination configs more for raw API usages (e.g. UI displaying list of namespaces, tables, etc.).

Reply:
Similar comment for pageSize, do we really need to expose configuration of this value in the library integration?

Reply:
I agree that we don't need this flag

Reply:
@geruh @jackye1995 @nastra 

Thanks for taking a look, I think its fine to remove the `REST_PAGINATION_ENABLED` flag.

 Im not sure though why we have to remove the configuration `REST_PAGE_SIZE` as how does the client (configure their engine such as spark) to set `pageSize` correctly?